libmount/src/utils.c

```diff
@@ -1342,6 +1348,199 @@ int mnt_tmptgt_cleanup(int old_ns_fd)
 #endif
 }
 
+#ifdef UL_HAVE_MOUNT_API
+static int write_id_mapping(idmap_type_t map_type, pid_t pid, const char *buf,
+			    size_t buf_size)
+{
+	int fd = -1, ret = -1, setgroups_fd = -1;
+	char path[PATH_MAX];
+
+	if (geteuid() != 0 && map_type == ID_TYPE_GID) {
+		snprintf(path, sizeof(path), "/proc/%d/setgroups", pid);
+
+		setgroups_fd = open(path, O_WRONLY | O_CLOEXEC | O_NOCTTY);
+		if (setgroups_fd < 0 && errno != ENOENT)
+			goto err;
+
+		if (setgroups_fd >= 0) {
+			ret = write(setgroups_fd, "deny\n", strlen("deny\n"));
```

Comment:
You can use write_all() from include/all-io.h to be more robust.

Reply:
Will do!